api/envoy/config/route/v3/route_components.proto

```diff
@@ -549,7 +550,33 @@ message CorsPolicy {
   core.v3.RuntimeFractionalPercent shadow_enabled = 10;
 }
 
-// [#next-free-field: 37]
+message HeaderRewriteLiteral {
+  // String match can use prefix match and present match. In case of prefix match, the matched prefix part of
+  // the header's value is substituted. In case of present match the whole value of the header is overwritten.
```

Comment:
What are the semantics for the other string match types? Does capture group semantics apply? 

I wonder if we want a `StringMatchAndSubstitute` that is analogous to `RegexMatchAndSubstitute`.